components/activity_and_insights/insights/top_channels/top_channels_line_chart/top_channels_line_chart.tsx

```diff
@@ -0,0 +1,205 @@
+                borderColor: theme.dndIndicator,
+                pointBackgroundColor: theme.dndIndicator,
+                pointBorderColor: theme.dndIndicator,
+                backgroundColor: 'transparent',
+                pointRadius: 0,
+                hoverBackgroundColor: theme.dndIndicator,
+                label: topChannels[3].display_name,
+                data: data.values[topChannels[3].id],
+                hitRadius: 10,
+            });
+        }
+
+        if (topChannels[4]) {
+            dataset.push({
+                fillColor: theme.newMessageSeparator,
+                borderColor: theme.newMessageSeparator,
+                pointBackgroundColor: theme.newMessageSeparator,
+                pointBorderColor: theme.newMessageSeparator,
+                backgroundColor: 'transparent',
+                pointRadius: 0,
+                hoverBackgroundColor: theme.newMessageSeparator,
+                label: topChannels[4].display_name,
+                data: data.values[topChannels[4].id],
+                hitRadius: 10,
+            });
+        }
+
+        return {
+            datasets: dataset,
+            labels: data.labels,
+        };
+    }, []);
+
+    return (
+        <LineChart
+            title={
+                <FormattedMessage
+                    id='analytics.system.totalBotPosts'
+                    defaultMessage='Top Channels'
+                />
+            }
+            id='totalPostsPerChannel'
+            options={{
+                responsive: true,
+                hover: {
+                    mode: 'point',
+                },
+                maintainAspectRatio: false,
+                scales: {
+                    xAxes: [{
+                        gridLines: {
+                            drawOnChartArea: false,
+                        },
+                        ticks: {
+                            callback(val, index) {
+                                // Hide every 4th tick label for 28 day time frame
+                                if (timeFrame === TimeFrames.INSIGHTS_28_DAYS) {
+                                    return index % 4 === 0 ? val : '';
+                                }
+                                return val;
+                            },
+                        },
+                    }],
+                    yAxes: [{
+                        gridLines: {
+                            drawOnChartArea: true,
+                        },
+                        ticks: {
+                            maxTicksLimit: 5,
+                            beginAtZero: true,
+                        },
+                    }],
+                },
+                tooltips: {
+                    callbacks: {
+                        label(tooltipItem, data) {
+                            const index = tooltipItem.datasetIndex;
+                            if (typeof index !== 'undefined' && data.datasets && data.datasets[index] && data.datasets[index].label) {
```

Comment:
I think this is equivalent...
```suggestion
                            if (typeof index !== 'undefined' && data?.datasets[index]?.label) {
```